hbase-server/src/main/java/org/apache/hadoop/hbase/replication/HBaseReplicationEndpoint.java

```diff
@@ -63,7 +121,7 @@ protected synchronized void disconnect() {
    * A private method used to re-establish a zookeeper session with a peer cluster.
    * @param ke
    */
-  protected void reconnect(KeeperException ke) {
+  private void reconnect(KeeperException ke) {
```

Comment:
Can we keep all accessors to ZKWatcher `protected` or at least, `package private`? In some cases, custom HBaseReplicationEndpoint implementations may need to access those.

Reply:
This class is IA.Private, so it is not expected to be used by users. You can stop someone change this method this time but you can not stop them all the time. If you think this class should be used by users  so they can implement replication endpoint easier, let's promote this class to IA.LimitedPrivate?

Reply:
Now only ReplicationEndpoint is IA.LimitedPrivate. So if one user want to use these logics, it is better to copied these code to their implementation.

Reply:
> So if one user want to use these logics, it is better to copy these code to their implementation.

It's not that simple, as it would actually require reimplement all other methods currently referring these private ones, leading to a reimplementation of the whole class. 

I agree with @Apache9 , if we are to guarantee a certain level of accessibility here, we should rather discuss about the interfaces audiences. I guess this would be rather a topic for a separate jira/PR, don't want to deviate this current one from it's original purpose.

Reply:
There is a issue about this, see HBASE-15982.